dashboard/config/libraries/variableLog.interpreted.js

```diff
@@ -1,78 +1,110 @@
+function buildVariableLog() {
+  text("Building variable log...",0,15);
+  var varLog = {};
+  var start = Object.keys(window).indexOf("studentVarToken") + 1;
+  var end = Object.keys(window).length;
   var index;
-  for(var i=start;i<end;i++){
-    if(window[Object.keys(window)[i]] != undefined){
-      if (typeof window[Object.keys(window)[i]] == "number" || typeof window[Object.keys(window)[i]] == "string"|| typeof window[Object.keys(window)[i]] == "boolean") {
-        varLog[Object.keys(window)[i]] = window[Object.keys(window)[i]];
-      }
+  for (var i = start; i < end; i++) {
+    // Blockly functions and behaviors should be excluded. 
+    if (
+      typeof window[Object.keys(window)[i]] == "number" ||
+      typeof window[Object.keys(window)[i]] == "string" ||
+      typeof window[Object.keys(window)[i]] == "boolean"
+    ) {
+      varLog[Object.keys(window)[i]] = window[Object.keys(window)[i]];
     }
   }
+  return varLog;
+}
 
-  varWatchers(varLog);
+function detectVariableLogChange() {
+  var result = false;
+  if(varLog && previousVarLog){
+    for (var property in varLog) {
+      if (varLog[property] !== previousVarLog[property]) {
+        //window[Object.keys(window)[Object.keys(window).indexOf(property)]]
+        result = true;
+      }
+    }
+  }
+  return result;
+}
 
-  if(validationProps){//is Sprite Lab
-    if(!validationProps.previous){
-      validationProps.previous = {};
+function updateVariableLog() {
+  for (var property in varLog) {
+    if (varLog[property] !== window[Object.keys(window)[Object.keys(window).indexOf(property)]]) {
+      varLog[property] = window[Object.keys(window)[Object.keys(window).indexOf(property)]];
     }
-    detectVarChange();
-    validationProps.previous.varLog=JSON.parse(JSON.stringify(varLog));
   }
+}
 
+function storeVariableLogforPrevious() {
+  previousVarLog = JSON.parse(JSON.stringify(varLog));
 }
 
-function varWatchers(varLog){
-  for(var key in varLog){
+
+// Draw a set of on-screen watchers, ie. show each student variable name
+// and it's value, in a drawn element on the canvas. (Prototype)
+function varWatchers(varLog) {
+  for (var key in varLog) {
     var index = Object.keys(varLog).indexOf(key);
     var x = 5;
     var y = 32;
-    watcher(key, varLog[key], index, x, y);
+    drawWatcher(key, varLog[key], iqndex, x, y);
   }
 }
 
-function watcher(label, value, index, x, y){
-  if(!value&&value!=0&&value!=''){
-    value = 'undefined';
+function drawWatcher(label, value, index, x, y) {
+  if (!value && value != 0 && value != "") {
+    value = "undefined";
   }
-  //rect(fontSize/2,fontSize/2+index*fontSize*2);
-  var fontSize=15;
+  var fontSize = 15;
   textSize(fontSize);
-  textAlign(LEFT,CENTER);
-  labelX = x + 5;
-  valueX = x + 15 + textWidth(label);
-  textY = y + (index+0.5) * fontSize*2;
+  textAlign(LEFT, CENTER);
+  var labelX = x + 5;
+  var valueX = x + 15 + textWidth(label);
+  var textY = y + (index + 0.5) * fontSize * 2;
```

Comment:
I'm slightly confused by this calculation.  Why would we add `0.5` to `index` which is presumably an integer?